benchmarks/huggingface.py

```diff
@@ -354,7 +444,87 @@ def forward_and_backward_pass(self, mod, inputs, collect_outputs=True):
         return None
 
 
+def refresh_model_names():
+    import transformers.utils.fx as hf_fx
+
+    family = dict()
+    lm_seen = set()
+    family_seen = set()
+    for cls_name in hf_fx._SUPPORTED_MODELS:
+
+        if "For" not in cls_name:
+            continue
+
+        model_cls = get_module_cls_by_model_name(cls_name)
+
+        # TODO: AttributeError: '*Config' object has no attribute 'vocab_size'
+        if model_cls in [
+            CLIPModel,
+            CLIPVisionModel,
+            SwinForImageClassification,
+            SwinForImageClassification,
+            SwinForMaskedImageModeling,
+            SwinModel,
+            ViTForImageClassification,
+            ViTForMaskedImageModeling,
+            ViTModel,
+        ]:
+            continue
+
+        # TODO: AssertionError: Padding_idx must be within num_embeddings
+        if model_cls in [MarianForCausalLM, MarianMTModel, MarianModel]:
+            continue
+
+        # TODO: "model is not supported yet" from HFTracer
+        if model_cls in [HubertForSequenceClassification]:
+            continue
+
+        # TODO: shape mismatch in loss calculation
+        if model_cls in [LxmertForQuestionAnswering]:
+            continue
+
+        family_name = cls_name.split("For")[0]
+        if family_name not in family:
+            family[family_name] = []
+        if cls_name.endswith(("MaskedLM", "CausalLM")) and family_name not in lm_seen:
+            family[family_name].append(cls_name)
+            lm_seen.add(family_name)
+        elif (
+            cls_name.endswith(
+                ("SequenceClassification", "ConditionalGeneration", "QuestionAnswering")
+            )
+            and family_name not in family_seen
+        ):
+            family[family_name].append(cls_name)
+            family_seen.add(family_name)
+        elif cls_name.endswith("ImageClassification"):
+            family[family_name].append(cls_name)
+
+    chosen_models = set()
+    for members in family.values():
+        chosen_models.update(set(members))
+
+    chosen_models.update(set(EXTRA_MODELS.keys()))
+    filename = "huggingface_models_list.txt"
+    if os.path.exists("benchmarks"):
+        filename = "benchmarks/" + filename
+    for model_name in sorted(chosen_models):
+        try:
+            subprocess.check_call(
+                [sys.executable]
+                + sys.argv
+                + ["--find-batch-sizes"]
+                + [f"--only={model_name}"]
+                + [f"--output={filename}"]
+            )
+        except subprocess.SubprocessError:
+            print("ERROR")
+
+
 if __name__ == "__main__":
-    logging.basicConfig(level=logging.WARNING)
-    warnings.filterwarnings("ignore")
-    main(HuggingfaceRunner())
+    if REFRESH_MODEL_NAMES and "--find-batch-sizes" not in sys.argv:
```

Comment:
How does this ever get triggered? It seems like REFRESH_MODEL_NAMES is hardcoded above.